node-util/sbin/oo-idler-stats

```diff
@@ -144,40 +143,65 @@ def get_app_type(app_path):
     except:
         return "unknown"
 
+app_name = ""
+cartridges_list = {}
 
 def validate(uuid):
     """ given a user's uuid, verify an app exists. """
     app_path = "/var/lib/openshift/%s" % uuid
 
-    # Is there a /git directory?
-    if not os.path.isdir(app_path + "/git"):
-        return "Path %s/git not found." % app_path
-
-    # Is there something inside the /git directory?
-    dirs = os.listdir(app_path + "/git")
-    if len(dirs) == 0:
-        return "Path %s/git is empty." % app_path
-
-    # Under /git should be appname.git
-    app_name = None
-    for d in dirs:
-        if re.search('\.git$', d):
-            app_name = d.replace(".git", "")
-
-    # Is there stuff inside git/appname.git?
-    gitfiles = os.listdir(app_path + "/git/" + app_name + ".git")
-    if len(gitfiles) == 0:
-        return "Path %s/git/%s.git is empty." % (app_path, app_name)
-
-    # Chances are good that this is a git repo. What's the last commit?
-    last_commit = run('git show HEAD | grep Date | sed "s/Date:   //"',
-                      app_path + "/git/" + app_name + ".git")[1][0]
+    global app_name
+    git_repo = False
+    last_commit = "N/A"
 
     # do any cartridges exists?
     types = cartridge_types(app_path)
     if 0 >= len(types):
         return "Path %s does not contain any cartridges." % app_path
 
+    # Loop through all cartridges on gear
+    for cart in types:
+        # Check if cartridge has been checked for cartridge type before
+        if cart in cartridges_list:
+            # Flag git_repo if this cartridge should have a git repo
+            if cartridge_types.get(cart) == true:
+                git_repo = True
+                break
+        else:
+            cmd = " ".join(["cat",os.path.join(app_path, cart, "metadata/manifest.yml"),
+                                               "| grep web_framework"])
+            ignore, result = run(cmd)
```

Comment:
It's more "pythonic" to use "_" instead of "ignore" for placeholders when unpacking tuples.